daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -260,10 +260,6 @@ class Compiler(var validateDFDLSchemas: Boolean = true)
     tunablesObj = tunablesObj.setTunables(tunables)
   }
 
-  def resetTunables(): Unit = {
```

Comment:
Just above this is a call to ExternalVariablesLoader.getVariables() which passes ina  tunablesObj.  Does that need to change too? Seems odd that variable stuff is getting tunable information from the Compiler, and not a DataProcessor, but mabye that happens so early it needs to?

Reply:
It's all the same tunables "stuff". The compiler constructs the tunablesObj but it can be modified. The DP gets it from the compiler post modifications. it gets serialized and back when the DP is serialized. Then it can be modified again. Finally that gets tucked into the PState/UState for use at runtime.

Reply:
Ah yeah, this makes perfect sense. If setExternalDFDLVariables is called on the Compiler it makes sense to use the tunables from the compiler. As long as there is no weird behavior if you later chnage the tunable.

Reply:
Ok, so the reason that the tunables are passed to the external variables getVariables call, is because the getVariables call has to resolve variable names, and the resolver requires the tunables so that it can have access to unqualifiedPathStepPolicy, which is necessary if what you are resolving is a path step.

For variable resolving, I think we can hard wire this to a particular value (because it doesn't matter), and then the getVariables call won't need a tunable argument.